packages/core/src/transport/flushController.spec.ts

```diff
@@ -0,0 +1,246 @@
+      expect(flushSpy).toHaveBeenCalled()
+    })
+
+    it('flush reason should be the page exit reason', () => {
+      flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+      flushController.didAddMessage()
+      pageExitObservable.notify({ reason: 'before_unload' })
+      expect(flushSpy.calls.first().args[0].reason).toBe('before_unload')
+    })
+
+    it('does not notify if no message was added', () => {
+      pageExitObservable.notify({ reason: 'before_unload' })
+      expect(flushSpy).not.toHaveBeenCalled()
+    })
+
+    it('notifies when the page is exiting even if no message have been fully added yet', () => {
+      flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+      pageExitObservable.notify({ reason: 'before_unload' })
+      expect(flushSpy).toHaveBeenCalled()
+    })
+  })
+
+  describe('bytes limit', () => {
+    it('notifies when the bytes limit is reached after adding a message', () => {
+      flushController.willAddMessage(BYTES_LIMIT)
+      flushController.didAddMessage()
+      expect(flushSpy).toHaveBeenCalled()
+    })
+
+    it('flush reason should be "bytes_limit"', () => {
+      flushController.willAddMessage(BYTES_LIMIT)
+      flushController.didAddMessage()
+      expect(flushSpy.calls.first().args[0].reason).toBe('bytes_limit')
+    })
+
+    it('notifies when the bytes limit will be reached before adding a message', () => {
+      flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+      flushController.didAddMessage()
+      flushController.willAddMessage(BYTES_LIMIT)
+      expect(flushSpy).toHaveBeenCalled()
+    })
+
+    it('does not take removed messages into account', () => {
+      flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+      flushController.didAddMessage()
+      flushController.didRemoveMessage(SMALL_MESSAGE_BYTE_COUNT)
+
+      flushController.willAddMessage(BYTES_LIMIT - SMALL_MESSAGE_BYTE_COUNT)
+      flushController.didAddMessage()
+
+      expect(flushSpy).not.toHaveBeenCalled()
+    })
+
+    it('does not notify when the bytes limit will be reached if no message was added yet', () => {
+      flushController.willAddMessage(BYTES_LIMIT)
+      expect(flushSpy).not.toHaveBeenCalled()
+    })
+
+    it('resets the current bytes count once flushed', () => {
+      flushController.willAddMessage(BYTES_LIMIT)
+      flushController.didAddMessage()
+
+      flushSpy.calls.reset()
+
+      flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+      flushController.didAddMessage()
+      expect(flushSpy).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('messages limit', () => {
+    it('notifies when the messages limit is reached', () => {
+      for (let i = 0; i < MESSAGES_LIMIT; i += 1) {
+        flushController.willAddMessage(SMALL_MESSAGE_BYTE_COUNT)
+        flushController.didAddMessage()
+      }
+      expect(flushSpy).toHaveBeenCalled()
+    })
+
+    it('flush reason should be "bytes_limit"', () => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏should not we have a `messages_limit` flush reason?